### PR TITLE
Framework: Refactor and remove FollowsList emitter store

### LINKED ISSUE #48405: Reduxify: FollowList and FollowListSite
A couple of simple `EventEmitter` stores that should be relatively straightforward to port ([source](https://github.com/Automattic/wp-calypso/blob/4537033c6794449ce0a10b69d72daf5fa658d553/client/lib/follow-list/index.js)).

SInce they have a single usage in Stats, we might as well use component state and lifecycle instead of Redux.

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR refactors the `FollowsList` into a simpler, direct approach, and removes the follows list emitter store that we were previously using. At first I thought about using Redux, but then I noticed that the usage is so simple that it is just not worth all the boilerplate.

#### Testing instructions

* Go to `/stats/insights/:site` for a site that has a bunch of visitors and followers.
* Observe the "Comments" and "Followers" sections at the bottom.
* Verify follow / unfollow buttons that appear for each user in those sections still work properly.

#### Notes

ESLint warnings are expected - original files contain them and they are not being introduced in this PR.

This PR fixes #48405.

### CODE REVIEW

File: client/my-sites/stats/stats-list/action-follow.jsx [hunk 1]
```diff
@@ -2,61 +2,84 @@
...
+import { gaRecordEvent } from 'calypso/lib/analytics/ga';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { getSiteDomain } from 'calypso/state/sites/selectors';
+import { requestSiteStats } from 'calypso/state/stats/lists/actions';
 
-const debug = debugFactory( 'calypso:stats:action-follow' );
-
-const StatsActionFollow = createReactClass( {
-	displayName: 'StatsActionFollow',
+class StatsActionFollow extends React.Component {
```
Comment (flootr) [inline client/my-sites/stats/stats-list/action-follow.jsx:19]: Love this refactor!

File: client/my-sites/stats/stats-list/action-follow.jsx [hunk 2]
```diff
@@ -2,61 +2,84 @@
...
 
-		if ( ! site.is_following ) {
+		if ( ! this.state.isFollowing ) {
 			gaEvent = 'Follow';
-			site.follow();
+			wpcom
+				.site( this.props.siteId )
+				.follow()
+				.add( { source: config( 'readerFollowingSource' ) } )
+				.then( this.refreshFollowers );
```
Comment (flootr) [inline client/my-sites/stats/stats-list/action-follow.jsx]: What happens if the request fails? Should we add a catch and reset `isFollowing` in that case?

File: client/my-sites/stats/stats-list/action-follow.jsx [hunk 3]
```diff
@@ -2,61 +2,84 @@
...
+				isFollowing: true,
+			} );
 		} else {
 			gaEvent = 'Unfollow';
-			site.unfollow();
+			wpcom
+				.site( this.props.siteId )
+				.follow()
+				.del( { source: config( 'readerFollowingSource' ) } )
+				.then( this.refreshFollowers );
```
Comment (flootr) [inline client/my-sites/stats/stats-list/action-follow.jsx]: I guess refetching the whole list was intentional but it differs in a significant way from prod where it blinks/flashes when new data is fetched. Does it make sense to do that update silently?

File: client/my-sites/stats/stats-list/action-follow.jsx (hunk 1 above)
Comment (tyxla) [reply client/my-sites/stats/stats-list/action-follow.jsx:19]: Me too! 😉

File: client/my-sites/stats/stats-list/action-follow.jsx (hunk 2 above)
Comment (tyxla) [reply client/my-sites/stats/stats-list/action-follow.jsx]: Good call - it's good to not be that naive. Addressed in https://github.com/Automattic/wp-calypso/pull/49825/commits/3febfb6062febb95746955e935d4e21bb7f1de2b.

File: client/my-sites/stats/stats-list/action-follow.jsx (hunk 3 above)
Comment (tyxla) [reply client/my-sites/stats/stats-list/action-follow.jsx]: I gave this a bit more thought and actually, we don't need to request all those stats just for the "is following" flag. Now that we handle errors, we should be good to rely on component state. Changed to be silent as of https://github.com/Automattic/wp-calypso/pull/49825/commits/d8fe8edca45765bdc05221c7cdddd93c014ccb11

**Comment by tyxla** [comment]
Thanks a bunch for the thorough review @flootr! I've addressed your feedback. Would you mind giving it another look when you get a chance?

**Review by flootr — APPROVED** [review]
Works great! :shipit:

**Comment by tyxla** [comment]
Thanks again, Markus! 🙌

**Pull request merged**